python/profiles/profiles.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+import sys
+import os
+from excludelist import excludelist
+from optiprofiler import set_cutest_problem_options, find_cutest_problems, run_benchmark
+import argparse
+from time import time
+
+
+def pdfo_uobyqa(fun, x0):
+    from pdfo import pdfo
+
+    res = pdfo(fun, x0, method='uobyqa')
+    return res.x
+
+
+def prima_uobyqa(fun, x0):
+    from prima import minimize
+    
+    res = minimize(fun, x0, method='uobyqa')
+    return res.x
+
+
+def pdfo_newuoa(fun, x0):
+    from pdfo import pdfo
+
+    res = pdfo(fun, x0, method='newuoa')
+    return res.x
+
+
+def prima_newuoa(fun, x0):
+    from prima import minimize
+
+    res = minimize(fun, x0, method='newuoa')
+    return res.x
+
+
+def pdfo_bobyqa(fun, x0, lb, ub):
+    from pdfo import pdfo
+    from scipy.optimize import Bounds
+
+    bounds = Bounds(lb, ub)
+    res = pdfo(fun, x0, method='bobyqa', bounds=bounds)
+    return res.x
+
+
+def prima_bobyqa(fun, x0, lb, ub):
+    from prima import minimize
+    from scipy.optimize import Bounds
+
+    bounds = Bounds(lb, ub)
+    res = minimize(fun, x0, method='bobyqa', bounds=bounds)
+    return res.x
+
+
+def pdfo_lincoa(fun, x0, lb, ub, a_ub, b_ub, a_eq, b_eq):
+    from pdfo import pdfo
+    from scipy.optimize import Bounds, LinearConstraint
+
+    # Until we include the elimination of linear equality constraints in PRIMA,
+    # we disable it in PDFO to make the comparison fair.
+    options = {'eliminate_lin_eq': False}
+    bounds = Bounds(lb, ub)
+    constraints = []
+    if b_ub.size > 0:
```

Comment:
Would the code still work if we remove `if b_ub.size > 0:` and `if b_eq.size > 0`? If not, who will stop working, SciPy or PRIMA? If SciPy allowst it, PRIMA should be able to handle it.

Reply:
The  same question is applicable below whenever the sizes are checked.

Reply:
`LinearConstraint` is now the same object in PRIMA and SciPy. PRIMA re-exports it for convenience of the interface (i.e. so that the user can write `from PRIMA import LinearConstraint` instead of having a separate line to grab it from SciPy).

I tried locally, if `b_ub` is of size 0 an exception is thrown due to the inability to broadcast it.

Reply:
Is the exception from SciPy? In other words, its SciPy's incapability, not ours, right? If it is the case, the no problem. 

Or let me put it in this way: if we use `scipy.optimize.minimize` to solve some problem, do we need to handle `b_up.size = 0` separately?

Reply:
Yes the exception is from SciPy.